grammars/jq.cson

```diff
@@ -136,7 +180,7 @@
       }
     ]
   'operator':
-    'match': '\\.\\.?|\\?|==?|//=?|\\|=?|\\+=?|-=?|\\*=?|/=?|%=?|!=|<=?|>=?'
+    'match': '\\.\\.?|//\\?|\\?|==?|//=?|\\|=?|\\+=?|-=?|\\*=?|/=?|%=?|!=|<=?|>=?'
```

Comment:
Do the other operators in this pattern include literal question-marks, or are they quantifiers?

Because `|=?|` is meaningless (it matches something that's either `=`, or nothing at all). I assume that `>=?` and friends might be trying to match `>` as well as `>=`...

Anyway, I'd format the pattern like this (but that's really just personal preference):

<details><summary><b>Show</b></summary>

~~~regexp
match: """(?x)
	( !=
	| %=?
	| -=?
	| //=?
	| //\?
	| /=?
	| <=?
	| ==?
	| >=?
	| \*=?
	| \+=?
	| \.\.?
	| \?
	| \|=?
	)
"""
~~~

</details>

Always remember to sort longer patterns before shorter ones when they're tied for alphabetisation:

```regexp
(if|ifelse)   # Bad

(ifelse|if)   # Good
```

This lessens the risk of matching only half of `ifelse`, because `if` appears first in the group, and is matched at the start of an `ifelse` token.

Reply:
> Do the other operators in this pattern include literal question-marks, or are they quantifiers?
> 
> Because `|=?|` is meaningless (it matches something that's either `=`, or nothing at all). I assume that `>=?` and friends might be trying to match `>` as well as `>=`...

I actually got it wrong, is should be `?//`. It's "Destructuring Alternative Operator" which allows to list several object destructing patterns and the first one to match will be used. So not a quantifier.

The only other question-mark is the `?` operator which is a short hand for `try`.

> Anyway, I'd format the pattern like this (but that's really just personal preference):

Yes think i would prefer that also, very hard to see what going on in that list.

> Always remember to sort longer patterns before shorter ones when they're tied for alphabetisation:
> 
> ```regexp
> (if|ifelse)   # Bad
> 
> (ifelse|if)   # Good
> ```
> 
> This lessens the risk of matching only half of `ifelse`, because `if` appears first in the group, and is matched at the start of an `ifelse` token.

Ok good point